packages/server-core/src/media/storageprovider/s3.storage.ts

```diff
@@ -367,16 +385,15 @@ export class S3Provider implements StorageProviderInterface {
     const listResponse = await this.listObjects(oldFilePath, true)
 
     const result = await Promise.all([
-      ...listResponse.Contents.map(async (file) =>
-        this.provider
-          .copyObject({
-            ACL: 'public-read',
-            Bucket: this.bucket,
-            CopySource: `/${this.bucket}/${file.Key}`,
-            Key: path.join(newFilePath, file.Key.replace(oldFilePath, ''))
-          })
-          .promise()
-      )
+      ...listResponse.Contents.map(async (file) => {
+        const input = {
+          ACL: 'public-read',
+          Bucket: this.bucket,
+          CopySource: `/${this.bucket}/${file.Key}`,
+          Key: path.join(newFilePath, file.Key.replace(oldFilePath, ''))
+        }
+        const command = new CopyObjectCommand(input)
```

Comment:
It doesn't look like this command is actually sent. It needs to be, and the response needs to be returned.